pennylane/templates/subroutines/commuting_evolution.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the CommutingEvolution template.
+"""
+# pylint: disable-msg=too-many-arguments
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+
+
+class CommutingEvolution(Operation):
+    r"""Applies the time-evolution operator for a Hamiltonian expressed as a linear combination
+    of mutually commuting Pauli words.
+
+    A commuting Hamiltonian is of the form
+
+    .. math:: H \ = \ \displaystyle\sum_{j} c_j P_j,
+
+    where :math:`P_j` are mutually commutative Pauli words and :math:`c_j` are real coefficients.
+    The time-evolution under a commuting Hamiltonian is given by a unitary of the form
+
+    .. math::
+
+        U(t) \ = \ e^{-i H t} \ = \exp(-i t \displaystyle\sum_j c_j P_j) =
+        \displaystyle\prod_j \exp(-i t c_j P_j).
+
+    If the Hamiltonian has a small number of unique eigenvalues, partial derivatives of observable
+    expectation values, i.e.
+
+    .. math:: \langle 0 | W(t)^\dagger O W(t) | 0 \rangle,
+
+    where :math:`W(t) = V U(t) Y` for some :math:`V` and :math:`Y`, taken with respect to
+    :math:`t` may be efficiently computed through generalized parameter shift rules. rules. When
+    initialized, this template will automatically compute the parameter-shift rule if given the
+    Hamiltonian's eigenvalue frequencies, i.e., the unique positive differences between
+    eigenvalues.
+
+    .. warning::
+
+       This template uses the :class:`~.ApproxTimeEvolution` operation with `n=1` in order to
+       implement the time evolution, as a single-step Trotterization is exact for a commuting
+       Hamiltonian. If the input Hamiltonian contains Pauli words which do not commute, the
+       compilation of the time evolution operator to a sequence of gates will not equate to the
+       exact propagation under the given Hamiltonian. Furthermore, if the specified frequencies
+       do not correspond to the true eigenvalue frequency spectrum of the commuting Hamiltonian,
+       computed gradients will be incorrect in general.

```

Comment:
```suggestion
       Hamiltonian.
       
       - If the input Hamiltonian contains Pauli words which do not commute, the
         compilation of the time evolution operator to a sequence of gates will
         not equate to the exact propagation under the given Hamiltonian.
         
       - Furthermore, if the specified frequencies do not correspond to the
         true eigenvalue frequency spectrum of the commuting Hamiltonian,
         computed gradients will be incorrect in general.
```

Relatively minor, but while reading this my eyes kinda 'skimmed' this paragraph, and it might mean readers miss these important points. Turning them into bullet points might highlight them slightly more.